src/mtype.h

```diff
@@ -208,12 +216,13 @@ struct mtype {
         std::string death_drops;
         float luminance;           // 0 is default, >0 gives luminance to lightmap
         int hp;
-        std::vector<unsigned int> sp_freq;     // How long sp_attack takes to charge
+        // special attack frequencies and function pointers
+        std::unordered_map<std::string, mtype_special_attack> special_attacks;
+        std::vector<std::string> special_attacks_names; // names of attacks, in json load order
```

Comment:
Having two containers in mtype (`special_attacks` and `special_attacks_names`) that store (at least partially) the same data is redundant. It seems the information about the load-order is required only when loading old saves.

In absences of any better (non-redundant) system, at least make `special_attacks_names` private and document that it is only to be used when loading old saves.